index.bs

```diff
@@ -11302,6 +11306,147 @@ The [=remote end steps=] given |session| and |command parameters| are:
 </div>
 
 
+## The extensions Module ## {#module-extensions}
+
+The <dfn export for=modules>extensions</dfn> module contains functionality for
+managing and interacting with extensions.
+
+### Definition ### {#module-extensions-definition}
+
+[=remote end definition=]
+
+<pre class="cddl remote-cddl">
+ExtensionsCommand = (
+  extensions.Install
+)
+</pre>
+
+[=local end definition=]
+
+<pre class="cddl local-cddl">
+ExtensionsResult = (
+  extensions.InstallResult
+)
+</pre>
+
+### Types ### {#module-extensions-types}
+
+#### The extensions.Extension Type #### {#type-extensions-Extension}
+
+<pre class="cddl local-cddl">
+extensions.ExtensionArchive = {
+   files: [+text],
+}
+</pre>
+
+<pre class="cddl remote-cddl local-cddl">
+extensions.Extension = {
+   installPath: extensions.ExtensionPath / extensions.ExtensionArchivePath / extensions.ExtensionBase64Encoded,
+   ? allowPrivateBrowsing: bool .default false,
+}
+
+extensions.ExtensionPath = {
+   type: "path",
+   path: text,
+}
+
+extensions.ExtensionArchivePath = {
+   type: "archive-path",
+   path: text,
+}
+
+extensions.ExtensionBase64Encoded = {
+   type: "base64",
+   value: text,
+}
+</pre>
+
+The <code>extensions.Extension</code> type represents an extension with its additional properties.
+
+<div algorithm>
+To <dfn>expand a install path spec</dfn> given |install path spec|:
```

Comment:
not sure if this should be a separate algorithm or something that should be inlined with the install command

Reply:
A separate algorithm is fine.

For the sake of concreteness, I suggest that we try to make this return a https://fs.spec.whatwg.org/#directory which is interpreted as the root of the extension.